### PR TITLE
Fcntl e0000025

### PR DESCRIPTION
**Pull request by clienthax**
This is likely not the best way to do this, but it seems to work 👀 

image

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 1]
```diff
@@ -1709,9 +1710,74 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
 		break;
 	}
 
-	case 0x00000025: // cellFsSdataOpenWithVersion
+	case 0xe0000025: // cellFsSdataOpenWithVersion
 	{
 		break;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1715]: NANI

**Review by AniLeo — COMMENTED** [review]
Not working as is for some reason, 0xe0000025 behaves exactly as master

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1715]: lol

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 2]
```diff
@@ -1709,9 +1710,74 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
...
+			break;
+		}
+
+		if (arg->_x4 != 0x10u || arg->_x8 != 0x28u)
+		{
+			sys_fs.error("sys_fs_fcntl(0xe0000025): invalid args (0x%x, 0x%x)", arg->_x4, arg->_x8);
+			break;
+		}
+
+		// Load mountpoint (doesn't support multiple // at the start)
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1731]: copypaste, irrelvant comment.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp (hunk 1 above)
Comment (clienthax) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1715]: Turns out the answer to how many Devs does it take to see a copy paste fail is 5

**Comment by clienthax** [comment]
> Not working as is for some reason, 0xe0000025 behaves exactly as master

Removed break :')

**Comment by clienthax** [comment]
Not too sure if this is actually functioning correctly, needs someone with more brains to check it over :D

**Comment by clienthax** [comment]
Looks like im not closing off the original file handle here, but not too sure on how to sort it..
image

**Comment by elad335** [comment]
> Looks like im not closing off the original file handle here, but not too sure on how to sort it..
> image

lv2_file::open_raw is what you are looking for, not sys_fs_open.

**Comment by clienthax** [comment]
Elad magic applied, Should be good for merge.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 3]
```diff
@@ -1709,9 +1710,54 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
 		break;
 	}
 
-	case 0x00000025: // cellFsSdataOpenWithVersion
+	case 0xe0000025: // cellFsSdataOpenWithVersion
 	{
-		break;
+
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1715]: remove this empty line

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 4]
```diff
@@ -1709,9 +1710,53 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
...
+		}
+
+		if (arg->_x4 != 0x10u || arg->_x8 != 0x28u)
+		{
+			sys_fs.error("sys_fs_fcntl(0xe0000025): invalid args (0x%x, 0x%x)", arg->_x4, arg->_x8);
+			break;
+		}
+
+		std::string_view vpath{ arg->name.get_ptr(), arg->name_size };
+		vpath = vpath.substr(0, vpath.find_last_not_of('\0') + 1);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1730]:
```suggestion
		vpath = vpath.substr(0, vpath.find_first_of('\0'));
```

**Review by elad335 — APPROVED** [review]

**Pull request merged**